src/webgpu/shader/execution/expression/call/builtin/textureStore.spec.ts

```diff
@@ -120,3 +120,377 @@ Parameters:
       .combine('C', ['i32', 'u32'] as const)
   )
   .unimplemented();
+
+// Texture width for dimensions >1D.
+// Sized such that mip level 2 will be at least 256 bytes/row.
+const kWidth = 256;
+
+// Returns the texture geometry based on a given number of texels.
+function getTextureSize(numTexels: number, dim: GPUTextureDimension, array: number): GPUExtent3D {
+  const size: GPUExtent3D = { width: 1, height: 1, depthOrArrayLayers: 1 };
+  switch (dim) {
+    case '1d':
+      size.width = numTexels;
+      break;
+    case '2d': {
+      const texelsPerArray = numTexels / array;
+      size.width = kWidth;
+      size.height = texelsPerArray / kWidth;
+      size.depthOrArrayLayers = array;
+      break;
+    }
+    case '3d':
+      size.width = kWidth;
+      size.height = numTexels / (2 * kWidth);
+      size.depthOrArrayLayers = 2;
+      break;
+  }
+  return size;
+}
+
+// WGSL declaration type for the texture.
+function textureType(dim: GPUTextureDimension): string {
+  return `texture_storage_${dim}<r32uint, write>`;
+}
+
+// Defines a function to convert linear global id into a texture coordinate.
+function indexToCoord(dim: GPUTextureDimension, type: string): string {
+  switch (dim) {
+    case '1d':
+      return `
+fn indexToCoord(id : u32) -> ${type} {
+  return ${type}(id);
+}`;
+      break;
+    case '2d':
+      return `
+fn indexToCoord(id : u32) -> vec2<${type}> {
+  return vec2<${type}>(${type}(id % width), ${type}(id / width));
+}`;
+      break;
+    case '3d':
+      return `
+fn indexToCoord(id : u32) -> vec3<${type}> {
+  const half = numTexels / depth;
+  let half_id = id % half;
+  return vec3<${type}>(${type}(half_id % width), ${type}(half_id / width), ${type}(id / half));
+}`;
+      break;
+  }
+  return ``;
+}
+
+// Mutates 'coords' to produce an out-of-bounds value.
+// 1D workgroups are launched so 'gid.x' is the linear id.
+function outOfBoundsValue(dim: GPUTextureDimension, type: string): string {
+  switch (dim) {
+    case '1d': {
+      if (type === 'i32') {
+        return `if gid.x % 3 == 0 {
+          coords = -coords;
```

Comment:
Deserves a comment: That coords has non-zero components. And that's true because this code is guarded in the enclosing shader by gid.x % 2 != 0.